src/uu/split/src/split.rs

```diff
@@ -409,22 +426,40 @@ impl Strategy {
 
 /// Parse the suffix type from the command-line arguments.
 fn suffix_type_from(matches: &ArgMatches) -> Result<(SuffixType, usize), SettingsError> {
-    if matches.value_source(OPT_NUMERIC_SUFFIXES) == Some(ValueSource::CommandLine) {
-        let suffix_start = matches.get_one::<String>(OPT_NUMERIC_SUFFIXES);
-        let suffix_start = suffix_start.ok_or(SettingsError::SuffixNotParsable(String::new()))?;
-        let suffix_start = suffix_start
-            .parse()
-            .map_err(|_| SettingsError::SuffixNotParsable(suffix_start.to_string()))?;
-        Ok((SuffixType::Decimal, suffix_start))
-    } else if matches.value_source(OPT_HEX_SUFFIXES) == Some(ValueSource::CommandLine) {
-        let suffix_start = matches.get_one::<String>(OPT_HEX_SUFFIXES);
-        let suffix_start = suffix_start.ok_or(SettingsError::SuffixNotParsable(String::new()))?;
-        let suffix_start = usize::from_str_radix(suffix_start, 16)
-            .map_err(|_| SettingsError::SuffixNotParsable(suffix_start.to_string()))?;
-        Ok((SuffixType::Hexadecimal, suffix_start))
-    } else {
-        // no numeric/hex suffix
-        Ok((SuffixType::Alphabetic, 0))
+    // Check if the user is specifying one or more than one suffix
+    // Any combination of suffixes is allowed
+    // Since all suffixes are setup with 'overrides_with_all()' against themselves and each other,
+    // last one wins, all others are ignored and will not match Some(ValueSource::CommandLine)
+    //
+    // Note: right now, this exact behavior cannot be handled by
+    // `ArgGroup` since `ArgGroup` considers a default value `Arg`
+    // as "defined".
+    match (
+        matches.value_source(OPT_NUMERIC_SUFFIXES) == Some(ValueSource::CommandLine),
+        matches.value_source(OPT_NUMERIC_SUFFIXES_SHORT) == Some(ValueSource::CommandLine),
+        matches.value_source(OPT_HEX_SUFFIXES) == Some(ValueSource::CommandLine),
+        matches.value_source(OPT_HEX_SUFFIXES_SHORT) == Some(ValueSource::CommandLine),
```

Comment:
Can we avoid `value_source`? If I recall correctly, we only have to use that if there is some default value, but I don't think these arguments have one.

Reply:
yeah ... considering they are all set to override each other and no default_value, can refactor